### PR TITLE
Auth test improvements

### PR DESCRIPTION
**Pull request by skisel-da**
This PR contains the following:
1) Introduce `ServiceCallContext` to provide more context fields into the call under the test
2) Additional auth tests to cover IdentityProviderConfigService

### CODE REVIEW

File: ledger/sandbox-on-x/src/test/it/scala/com/daml/ledger/platform/sandbox/auth/GetAdminUserWithGivenUserIdAuthIT.scala [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      stub(UserManagementServiceGrpc.stub(channel), context.token)
+        .getUser(GetUserRequest(userId))
+        .map(_.user.get)
+
+    for {
+      // create a normal users
+      (alice, _) <- createUserByAdmin(testId + "-alice")
+
+      // test that only admins can retrieve his own user and the newly created alice user
+      _ <- getUser("participant_admin")
```
Comment (pbatko-da) [inline ledger/sandbox-on-x/src/test/it/scala/com/daml/ledger/platform/sandbox/auth/GetAdminUserWithGivenUserIdAuthIT.scala:29]: This doesn't test that _only_ the admin can do that, does it?

File: ledger/sandbox-on-x/src/test/it/scala/com/daml/ledger/platform/sandbox/auth/ListUserRightsWithGivenAdminUserIdAuthIT.scala [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    def getRights(userId: String): Future[ListUserRightsResponse] =
+      stub(context.token).listUserRights(ListUserRightsRequest(userId))
+
+    for {
+      // create a normal users
+      (alice, _) <- createUserByAdmin(testId + "-alice")
+
+      // test that only admins can retrieve his own user and the newly created alice user
+      _ <- getRights("participant_admin")
```
Comment (pbatko-da) [inline ledger/sandbox-on-x/src/test/it/scala/com/daml/ledger/platform/sandbox/auth/ListUserRightsWithGivenAdminUserIdAuthIT.scala:30]: This doesn't test that only the admin can do that, does it?

**Review by pbatko-da — APPROVED** [review]

File: ledger/sandbox-on-x/src/test/it/scala/com/daml/ledger/platform/sandbox/auth/ListAuthenticatedUserRightsAuthIT.scala [hunk 3]
```diff
@@ -16,14 +16,15 @@ class ListAuthenticatedUserRightsAuthIT extends ServiceCallAuthTests {
   override def serviceCallName: String =
     "UserManagementService#ListUserRights(<authenticated-user>)"
 
-  override def serviceCallWithToken(token: Option[String]): Future[Any] =
-    stub(UserManagementServiceGrpc.stub(channel), token).listUserRights(ListUserRightsRequest())
+  protected def serviceCall(context: ServiceCallContext): Future[Any] =
+    stub(UserManagementServiceGrpc.stub(channel), context.token)
+      .listUserRights(ListUserRightsRequest())
```
Comment (mziolekda) [inline ledger/sandbox-on-x/src/test/it/scala/com/daml/ledger/platform/sandbox/auth/ListAuthenticatedUserRightsAuthIT.scala:21]: Aren't we supposed to populate the idp field?

File: ledger/sandbox-on-x/src/test/it/scala/com/daml/ledger/platform/sandbox/auth/ListUserRightsWithGivenAdminUserIdAuthIT.scala (hunk 2 above)
Comment (skisel-da) [reply ledger/sandbox-on-x/src/test/it/scala/com/daml/ledger/platform/sandbox/auth/ListUserRightsWithGivenAdminUserIdAuthIT.scala:30]: this test is actually unrelated to this PR, I will remove these 2 new classes

File: ledger/sandbox-on-x/src/test/it/scala/com/daml/ledger/platform/sandbox/auth/ListAuthenticatedUserRightsAuthIT.scala (hunk 3 above)
Comment (skisel-da) [reply ledger/sandbox-on-x/src/test/it/scala/com/daml/ledger/platform/sandbox/auth/ListAuthenticatedUserRightsAuthIT.scala:21]: well, spotted, though currently there is no test to try this out, it's not part of this PR yet :)

File: ledger/sandbox-on-x/src/test/it/scala/com/daml/ledger/platform/sandbox/auth/ListAuthenticatedUserRightsAuthIT.scala [hunk 4]
```diff
@@ -16,14 +16,15 @@ class ListAuthenticatedUserRightsAuthIT extends ServiceCallAuthTests {
...
 
   protected def expectRights(
-      token: Option[String],
+      context: ServiceCallContext,
       expectedRights: Vector[Right],
   ): Future[Assertion] =
-    serviceCallWithToken(token).map(assertResult(ListUserRightsResponse(expectedRights))(_))
+    serviceCall(context).map(assertResult(ListUserRightsResponse(expectedRights))(_))
 
   private def getRights(token: Option[String], userId: String) =
```
Comment (mziolekda) [inline ledger/sandbox-on-x/src/test/it/scala/com/daml/ledger/platform/sandbox/auth/ListAuthenticatedUserRightsAuthIT.scala:29]: should use context instead?

File: ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/ReadOnlyServiceCallAuthTests.scala [hunk 5]
```diff
@@ -25,25 +25,25 @@ trait ReadOnlyServiceCallAuthTests extends ServiceCallWithMainActorAuthTests {
       userPrefix: String,
       rights: Vector[proto.Right.Kind],
   ): Future[Any] =
-    createUserByAdmin(userPrefix + mainActor, rights.map(proto.Right(_)))
-      .flatMap { case (_, token) => serviceCallWithoutApplicationId(token) }
+    createUserByAdmin(userPrefix + mainActor, rights = rights.map(proto.Right(_)))
+      .flatMap { case (_, token) => serviceCall(ServiceCallContext(token)) }
```
Comment (mziolekda) [inline ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/ReadOnlyServiceCallAuthTests.scala:29]: includeApplicationId = false?

Comment (skisel-da) [reply ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/ReadOnlyServiceCallAuthTests.scala:29]: no, it wont work, this was a hack I removed, `serviceCallWithoutApplicationId` was a different impl from `serviceCallWithToken`

File: ledger/sandbox-on-x/src/test/it/scala/com/daml/ledger/platform/sandbox/auth/GetAdminUserWithGivenUserIdAuthIT.scala [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  override def serviceCall(context: ServiceCallContext): Future[Any] = {
+    val testId = UUID.randomUUID().toString
+
+    def getUser(userId: String): Future[User] =
+      stub(UserManagementServiceGrpc.stub(channel), context.token)
+        .getUser(GetUserRequest(userId))
+        .map(_.user.get)
+
+    for {
+      // create a normal users
```
Comment (mziolekda) [inline ledger/sandbox-on-x/src/test/it/scala/com/daml/ledger/platform/sandbox/auth/GetAdminUserWithGivenUserIdAuthIT.scala:25]:
```suggestion
      // create a normal user
```

File: ledger/sandbox-on-x/src/test/it/scala/com/daml/ledger/platform/sandbox/auth/GetAdminUserWithGivenUserIdAuthIT.scala [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+package com.daml.platform.sandbox.auth
+
+import com.daml.ledger.api.v1.admin.user_management_service._
+import io.grpc.{Status, StatusRuntimeException}
+
+import java.util.UUID
+import scala.concurrent.Future
+
+class GetAdminUserWithGivenUserIdAuthIT extends AdminServiceCallAuthTests {
```
Comment (mziolekda) [inline ledger/sandbox-on-x/src/test/it/scala/com/daml/ledger/platform/sandbox/auth/GetAdminUserWithGivenUserIdAuthIT.scala:12]: Is that a misnomer? I don't think we are "getting an admin user" in this test

File: ledger/sandbox-on-x/src/test/it/scala/com/daml/ledger/platform/sandbox/auth/ListUserRightsWithGivenAdminUserIdAuthIT.scala [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    with UserManagementAuth {
+
+  override def serviceCallName: String = "UserManagementService#ListUserRights(given-user-id)"
+
+  // admin and idp admin users are allowed to specify a user-id other than their own for which to retrieve a user
+  override def serviceCall(context: ServiceCallContext): Future[Any] = {
+    val testId = UUID.randomUUID().toString
+
+    def getRights(userId: String): Future[ListUserRightsResponse] =
+      stub(context.token).listUserRights(ListUserRightsRequest(userId))
```
Comment (mziolekda) [inline ledger/sandbox-on-x/src/test/it/scala/com/daml/ledger/platform/sandbox/auth/ListUserRightsWithGivenAdminUserIdAuthIT.scala:23]: idp field not populated, is that intentional?

Comment (skisel-da) [reply ledger/sandbox-on-x/src/test/it/scala/com/daml/ledger/platform/sandbox/auth/ListUserRightsWithGivenAdminUserIdAuthIT.scala]: this class is removed, the original class is fixed

File: ledger/sandbox-on-x/src/test/it/scala/com/daml/ledger/platform/sandbox/auth/GetAdminUserWithGivenUserIdAuthIT.scala (hunk 1 above)
Comment (skisel-da) [reply ledger/sandbox-on-x/src/test/it/scala/com/daml/ledger/platform/sandbox/auth/GetAdminUserWithGivenUserIdAuthIT.scala]: removed this class

File: ledger/sandbox-on-x/src/test/it/scala/com/daml/ledger/platform/sandbox/auth/GetAdminUserWithGivenUserIdAuthIT.scala (hunk 7 above)
Comment (skisel-da) [reply ledger/sandbox-on-x/src/test/it/scala/com/daml/ledger/platform/sandbox/auth/GetAdminUserWithGivenUserIdAuthIT.scala]: I removed this class

**Pull request merged**